mscore/playpanel.cpp

```diff
@@ -68,6 +68,9 @@ PlayPanel::PlayPanel(QWidget* parent)
       mgainSlider->setDclickValue1(seq->metronomeGain() - 10.75f);
       mgainSlider->setDclickValue2(seq->metronomeGain() - 10.75f);
 
+      volumeSlider->setDclickValue1(-20); // double click restores -40dB default
+      volumeSlider->setDclickValue2(-80); // right double click sets to mute
```

Comment:
Where did you find those values, and why -20 is actually -40? If you found them in another parts of the code, could perhaps the variables used there be reused here too, to avoid similar errors in future?

Reply:
Good questions without any very short answers that I can offer. The current way in which the gain sliders work (used in Mixer and Play Panel - not sure if elsewhere too) is, I'd suggest, rather ad hoc. You get gain to dB and back calculations both in the view (PlayPanel here) and in the custom volume slider (awl/volslider.cpp). The 20 difference is, I think, in the calculation made for the PlayPanel spin box. The volume slider itself uses a range from -60 to 20 (set in the VolSlider constructor).

A *good* thing to do would be to sort all of this out. Maybe the custom volume slider could manage all the calculations, so that there are in one place. Or maybe (or as well) the synthesiser could be changed to accept dB values and also provide a "default" dB value for sliders to read. Or add a setDefaultGain() to the synthesiser. It'll then signal all sliders that are listening to reset themselves. At the same time, this would also be a way of addressing the issue where the Mixer dB value is different from that of the PlayPanel's dB. (https://musescore.org/en/node/291453).

The *sorting all this out* option is not a small job. So it might be best to:
 *  defer this fix and do it right at some later date;  **or**
 *  go with this fix but expand the comment (and add a TODO:), deferring the *sorting out* to some later date; **or**
 * some other option...

Reply:
`volSpinBox->setValue(-40.0);`  and/or `static constexpr double N = 20.0 ;`?
And it not just the play panel, the synthesizer too goes to -20db on double click

Reply:
But in any case, you can't just define a value here.  it needs to be the actual default.  So you need to find where the actual default is being set and use the same defined constant in both places, so if the default ever changes, so does the double-click behavior.

Reply:
Probably looking at the places where that N gets used should help.

Reply:
Like `vol = ((N * std::log10(vol)) - N);` with vol being 0.1?

Reply:
@Jojo-Schmitz - thanks for pointing out synthesizer control

@MarcSabatella - I've created a defined constant for the default gain setting in MasterSynthesizer - all there was before was setting the initial value of gain to 0.1 - but after MasterSynthesizer had been created that value no recoverable within code. The default value is now available (including in decibels) to other objects. 

It seems to me that  the UI controls ought never to have been doing the log conversions and it certainly these should not have been appearing in multiple places. The revised PR puts the log conversions in just one place in MasterSynthesizer.

It turns out the mixer gain control didn't even have a double-click.

Any suggestions on the changes to MasterSynthesizer to be more idiomatic C++ would be appreciated. I sense that I've not quite got that right, although I believe it all runs fine. (I think I'm missing some const keywords that could be helpful and maybe there should be some use of statics there as well...)

I've set the the log mode to false for the gain slider used in the PlayPanel and SynthControl. There is still code in VolSlider and AbstractSlider that does log conversions. But it seems out of scope for this PR to remove that code from altogether, even though they might be a good thing to do.